### ISSUE TITLE
UseObjectInitializerDiagnostic should ignore self-referencing initializers

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#15402]
**Version Used**: 2017 RC 
```C#
class C {
	int x;
	C() {
		C c = new C();
		c.x = 1 - c.x;
	}
}
```

This incorrectly reports IDE0017, with the following broken codefix:

```C#
C c = new C() {
	x = 1 - c.x
};
```

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/14990

**Comment by SLaks** [comment]
It looks like http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/UseObjectInitializer/ObjectCreationExpressionAnalyzer.cs,126 should prevent this.

**Comment by SLaks** [comment]
Yes; exactly.
I couldn't tell when that line was added due to renames.

**Comment by CyrusNajmabadi** [comment]
No worries :) Thanks for the report!
